mindmeld/stemmers/stemmers.py

```diff
@@ -0,0 +1,73 @@
+import nltk
+from abc import ABC, abstractmethod
+
+
+class Stemmer(ABC):
+
+    def __init__(self, stemmer):
+        """
+        Args:
+            stemmer: The NLTK stemmer to be used
+        """
+        self._stemmer = stemmer
+
+    @abstractmethod
+    def stem_word(self, word):
+        """
+        Gets the stem of a word. For example, the stem of the word 'fishing' is 'fish'.
+
+        Args:
+            word (str): The word to stem
+
+        Returns:
+            list(str): A list of the stemmed version of a word.
+        """
+        pass
+
+
+class EnglishNLTKStemmer(Stemmer):
+
+    def __init__(self, stemmer=None):
+        del stemmer
+        super().__init__(nltk.stem.PorterStemmer())
+
+    def stem_word(self, word):
+        stem = word.lower()
+
+        if self._stemmer.mode == self._stemmer.NLTK_EXTENSIONS and word in self._stemmer.pool:
+            return [self._stemmer.pool[word]]
+
+        if self._stemmer.mode != self._stemmer.ORIGINAL_ALGORITHM and len(word) <= 2:
+            # With this line, strings of length 1 or 2 don't go through
+            # the stemming process, although no mention is made of this
+            # in the published algorithm.
+            return [word]
+
+        stem = self._stemmer._step1a(stem)
+        stem = self._stemmer._step1b(stem)
+        stem = self._stemmer._step1c(stem)
+        stem = self._stemmer._step5b(stem)
+
+        # if the stemmed cleaves off the whole token, just return the original one
+        if stem == '':
+            return [word]
+        else:
+            return [stem]
+
+
+class SpanishNLTKStemmer(Stemmer):
+
+    def __init__(self, stemmer=None):
+        del stemmer
+        super().__init__(nltk.stem.SnowballStemmer('spanish'))
```

Comment:
Let's make language an init arg. Also, how might a passed in stemmer be used?

Reply:
I would like the stemmers to be generic, ie., the developer can implement his/her stemmer given the contract specified in the abstract class `Stemmer`. Therefore, it make sense to have an in-built mindmeld spanish stemmer as it's own class, which the developer can choose to use (or not), rather than a param passed into a Stemming class that is opinionated on how spanish words should be stemmed.